Library/ViewModels/RootCommentCellViewModel.swift

```diff
@@ -0,0 +1,84 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+
+public protocol RootCommentCellViewModelInputs {
+  /// Call when bindStyles is called.
+  func bindStyles()
+
+  /// Call to configure with a Comment
+  func configureWith(comment: Comment)
+}
+
+public protocol RootCommentCellViewModelOutputs {
+  /// Emits author's badge for a comment.
+  var authorBadge: Signal<Comment.AuthorBadge, Never> { get }
+
+  /// Emits a url to the comment author's image.
+  var authorImageURL: Signal<URL, Never> { get }
+
+  /// Emits text containing author's fullname or username.
+  var authorName: Signal<String, Never> { get }
+
+  /// Emits text containing comment body.
+  var body: Signal<String, Never> { get }
+
+  /// Emits text  relative time the comment was posted.
+  var postTime: Signal<String, Never> { get }
+}
+
+public protocol RootCommentCellViewModelType {
+  var inputs: RootCommentCellViewModelInputs { get }
+  var outputs: RootCommentCellViewModelOutputs { get }
+}
+
+public final class RootCommentCellViewModel:
+  RootCommentCellViewModelType, RootCommentCellViewModelInputs, RootCommentCellViewModelOutputs {
+  public init() {
+    let comment = self.comment.signal.skipNil()
+      .map { comment in comment }
```

Comment:
hmm is this map doing anything?

Reply:
Nope! I did remove it, previously left it in because I saw another pr use that technique, but there's no need to do it.